bot/exts/events/hacktoberfest/_utils.py

```diff
@@ -0,0 +1,451 @@
+
+
+def author_mention_from_context(ctx: commands.Context) -> tuple[str, str]:
+    """Return stringified Message author ID and mentionable string from commands.Context."""
+    author_id = str(ctx.author.id)
+    author_mention = ctx.author.mention
+
+    return author_id, author_mention
+
+
+# Util functions for `.hacktoberfest issue`
+async def get_issues(ctx: commands.Context, option: str) -> Optional[dict]:
+    """Get a list of the python issues with the label 'hacktoberfest' from the Github api."""
+    if option == "beginner":
+        if (ctx.message.created_at.replace(tzinfo=None) - ctx.cog.cache_timer_beginner).seconds <= 60:
+            log.debug("using cache")
+            return ctx.cog.cache_beginner
+    elif (ctx.message.created_at.replace(tzinfo=None) - ctx.cog.cache_timer_normal).seconds <= 60:
+        log.debug("using cache")
+        return ctx.cog.cache_normal
+
+    if option == "beginner":
+        url = URL + '+label:"good first issue"'
+        if ctx.cog.cache_beginner is not None:
+            page = random.randint(1, min(1000, ctx.cog.cache_beginner["total_count"]) // 100)
+            url += f"&page={page}"
+    else:
+        url = URL
+        if ctx.cog.cache_normal is not None:
+            page = random.randint(1, min(1000, ctx.cog.cache_normal["total_count"]) // 100)
+            url += f"&page={page}"
+
+    log.debug(f"making api request to url: {url}")
+    async with ctx.bot.http_session.get(url, headers=ISSUES_REQUEST_HEADERS) as response:
+        if response.status != 200:
+            log.error(f"expected 200 status (got {response.status}) by the GitHub api.")
+            await ctx.send(
+                f"ERROR: expected 200 status (got {response.status}) by the GitHub api.\n"
+                f"{await response.text()}"
+            )
+            return None
+        data = await response.json()
+
+        if len(data["items"]) == 0:
+            log.error(f"no issues returned by GitHub API, with url: {response.url}")
+            await ctx.send(f"ERROR: no issues returned by GitHub API, with url: {response.url}")
+            return None
+
+        if option == "beginner":
+            ctx.cog.cache_beginner = data
+            ctx.cog.cache_timer_beginner = ctx.message.created_at.replace(tzinfo=None)
+        else:
+            ctx.cog.cache_normal = data
+            ctx.cog.cache_timer_normal = ctx.message.created_at.replace(tzinfo=None)
+
+        return data
+
+
+def format_issues_embed(issue: dict) -> discord.Embed:
+    """Format the issue data into a embed."""
+    title = issue["title"]
+    issue_url = issue["url"].replace("api.", "").replace("/repos/", "/")
+    # issues can have empty bodies, which in that case GitHub doesn't include the key in the API response
+    body = issue.get("body", "")
+    labels = [label["name"] for label in issue["labels"]]
+
+    embed = discord.Embed(title=title)
+    embed.description = body[:500] + "..." if len(body) > 500 else body
```

Comment:
Parsing the markdown seems too big of a thing for the scope of this PR. Perhaps @hedyhli would like to do this as one of their hacktoberfest PRs.

Reply:
Shouldn't the markdown already render like the GitHub webhook embeds?

Reply:
Yeah, most of the markdown is passed but HTML tags like `p`, `details` aren't used so often, so this feature is not really needed, but it would be good to have it.

Reply:
It would require flattening out (transforming HTML to plaintext) but I don't really think it's that much of a problem. Users should go to the GitHub issue page anyway to see other information like assignees and linked PRs. If you'd still like that you should open a issue and we can discuss about whether it should be added.